src/components/learnDetail/ConfirmModal.tsx

```diff
@@ -15,12 +15,31 @@ interface ConfirmModalProps {
   closeModal: (close: boolean) => void;
   setMemoHighlightId: (id: MemoHighlightId) => void;
   confirmModalText: ConfirmModalText;
+  clickedIndex: number;
+  setClickedIndex: (index: number) => void;
+  scriptIndexList: string[];
+  setScriptIndexList: (list: string[]) => void;
 }
 
 function ConfirmModal(props: ConfirmModalProps) {
-  const { closeModal, setMemoHighlightId, confirmModalText } = props;
+  const {
+    closeModal,
+    setMemoHighlightId,
+    confirmModalText,
+    clickedIndex,
+    setClickedIndex,
+    scriptIndexList,
+    setScriptIndexList,
+  } = props;
   const { mainText, subText, confirmText, cancelText } = confirmModalText;
 
+  const deleteIndex = (index: number) => {
+    const tempList = [...scriptIndexList];
+    tempList.splice(index, 1);
+    setScriptIndexList(tempList);
+    setClickedIndex(0);
+  };
```

Comment:
p4;
```suggestion
  const deleteIndex = (index: number) => {
    setScriptIndexList(scriptIndexList.filter((scriptIndex) => scriptIndex !== index));
    setClickedIndex(0);
  };
```
filter를 이용해서 임시 변수를 만들지 않는 방법도 있을 것 같습니다!

Reply:
여기 주석을 달아둘 걸 그랬네요 !!
스크립트 인덱스 삭제 API 연결하면서 tempList 사용한 37-39 라인은 제거하려고 합니다
조언 감사합니다 👍🏻